### PR TITLE
Test optimize memory leak, long loading optimize 

### PR DESCRIPTION
**Pull request by melnicenkovadik**

### CODE REVIEW

**Review by vitalii427 — CHANGES_REQUESTED** [review]
Undo wrapper refactoring

File: src/pages/Staking/Staking.js [hunk 1]
```diff
@@ -2,57 +2,45 @@ import React, { useEffect, useState } from 'react';
...
   useEffect(() => {
-    activate(connectorsByName.Injected);
-    const mounted = true;
-    if (mounted) {
-      getDataFromProvider();
-      if (ethereum?.isMetaMask) {
-        if (chainId !== +process.env.REACT_APP_CHAIN_ID) {
-          window.addEventListener('focus', changeNetwork);
-        }
+    console.log('Staking render useEffect');
```
Comment (Motzart) [inline src/pages/Staking/Staking.js:35]: remove this

File: src/services/staking.wrapper.js [hunk 2]
```diff
@@ -50,70 +52,90 @@ function parseFloatToBigNumber(str) {
   return ethers.utils.parseEther(math.round(mathBn, 18).toString());
 }
 
-class StakingWrapper {
+class StakingWrapperSingleton {
+  static privateInstance = null;
+
   constructor(providerOrSigner = null) {
+    console.log('StakingWrapper constructor');
```
Comment (Motzart) [inline src/services/staking.wrapper.js:59]: remove this

File: src/services/staking.wrapper.js [hunk 3]
```diff
@@ -50,70 +52,90 @@ function parseFloatToBigNumber(str) {
...
 
+  static getInstance() {
+    if (window.location.pathname !== '/') {
+      if (ethereum !== undefined) {
+        const provider = new providers.Web3Provider(ethereum);
+        const signer = provider.getSigner();
+        StakingWrapperSingleton.privateInstance = new StakingWrapperSingleton(
+          signer,
+        );
+        console.log('StakingWrapper(signer)');
```
Comment (Motzart) [inline src/services/staking.wrapper.js:105]: remove this

Comment (vitalii427) [reply src/services/staking.wrapper.js:105]: console.log ?

**Review by vitalii427 — APPROVED** [review]

**Pull request merged**